ocean_lib/models/data_token.py

```diff
@@ -40,6 +41,128 @@ class DataToken(ContractBase):
     OPF_FEE_PERCENTAGE = 0.001
     MAX_MARKET_FEE_PERCENTAGE = 0.001
 
+    # ============================================================
+    # reflect DataToken Solidity methods
+    def initialize(
+        self,
+        name: str,
+        symbol: str,
+        minter_address: str,
+        cap: int,
+        blob: str,
+        fee_collector_address: str,
+        from_wallet: Wallet,
+    ) -> str:
+        return self.send_transaction(
+            "initialize",
+            (name, symbol, minter_address, cap, blob, fee_collector_address),
+            from_wallet,
+        )
+
+    def mint(self, account_address: str, value_base: int, from_wallet: Wallet) -> str:
```

Comment:
@DMats I like what you did with the interface here. But since I added the tests and documented this part, I was wondering whether we could make all these functions more DRY.

It seems like all of them are self.send_transaction("`name_of_function`", `args_except_wallet`, `wallet`). Is there a way we could compound all of these somehow?

Reply:
@calina-c Thanks. 

I struggle to imagine how we could make these functions **more** DRY. Every `DataToken` method that calls `ContractBase.send_transaction` does so with different parameters. This same pattern is used throughout ocean.py, not just in `DataToken`.

Adding a compound data class (shown below) is possible, but would only reduce repetition when sending the same transaction twice. Since this is not a common use case, I don't see the need to explicitly support it. Users can use a [packed argument list](https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists) if they really want to send the same transaction twice.

```python
@dataclass
class Transaction
    """Class to hold the parameters for sending a transaction"""
    fn_name: str,
    fn_args: tuple,
    from_wallet: Wallet,
    raw_transaction: dict = None,
```

In conclusion, I don't think further changes are warranted. If I've misunderstood your question, please let me know.

Reply:
OK, let's leave it for some other time. It's outside of the scope either way. Great job!